_posts/2023-04-17-corewcf-1_4_0_preview.markdown

```diff
@@ -0,0 +1,145 @@
+        listenOptions.ConnectionBufferSize = 64 * 1024;
+    });
+});
+```
+
+There are a couple of items needing to be completed before named pipe support will come out of preview. The biggest one is that there's currently no WSDL support. The second one is when stopping the WebApplication, the channel close doesn't happen cleanly. Instead of informing the client that the channel is closing, the underlying named pipe connection is abruptly closed.
+
+### Queue based transports
+
+When discussing bringing MSMQ support in a CoreWCF issue, I mentioned I'd like to make something generic so that CoreWCF could support multiple queue transport protocols. There was a lot of positive response to this idea, so I wrote up a brief description of how that could look. Biroj Nayak (@birojnayak) from Amazon AWS took this description and after a few discussions wrote a detailed design document. Dmitry Maslov (@Ximik87) submitted a PR with an initial implementation of this base queue support along with an MSMQ implementation built on top of it. Jon Louie (@jonlouie) from Amazon AWS implemented the RabbitMQ transport on top of this. Biroj helped iter with help from Biroj due to the initial base queue support needing some changes to accomodate different API patterns of the RabbitMQ client library. There are two other queue transports coming soon. The first will be for Apache Kafka and is being contributed by Guillaume Delahaye (@g7ed6e). This will be part of and released by the CoreWCF project. The second will be for Azure Queue Storage which is being developed by Microsoft and will be released by them.  
+
+To use any of the queue transports, you need to add the generic queue support to your application services.
+
+```csharp
+var builder = WebApplication.CreateBuilder();
+builder.Services.AddServiceModelServices()
+                .AddQueueTransport();
+```
+
+#### MSMQ
+
+The CoreWCF project has a requirement to only depend on 3rd party packages which are backed by an open source foundation with a charter that ensures continued support of libraries if the maintainers step away from the project. A well known example in the .NET ecosystem is the [.NET Foundation](https://dotnetfoundation.org/). This is important to ensure that any potential security issues can be fixed in a timely manner. The current MSMQ implementation depends on a community owned release of a fork of the .NET Framework System.Messaging libraries which doesn't meet this requirement. The CoreWCF.MSMQ package will remain in a pre-release state until we have resolved this. We would encourage you to use the CoreWCF MSMQ transport as a stepping stone to moving to a more modern queue transport.  
+
+To use the MSMQ transport, in addition to adding the queue transport support, you also need to add Msmq support.
+
+```csharp
+var builder = WebApplication.CreateBuilder();
+builder.Services.AddServiceModelServices()
+                .AddQueueTransport()
+				.AddServiceModelMsmqSupport();
```

Comment:
nit: Spacing is inconsistent with `.AddServiceModelMsmqSupport()`. If this renders correctly, please ignore.

Reply:
That's a tabs vs spaces thing. There must be a mix of them. Thanks for spotting this.